modin/data_management/partitioning/partition_collections.py

```diff
@@ -102,7 +126,11 @@ def block_lengths(self):
             # The first column will have the correct lengths. We have an
             # invariant that requires that all blocks be the same length in a
             # row of blocks.
-            self._lengths_cache = [obj.length().get() for obj in self.partitions.T[0]]
+            self._lengths_cache = (
+                [obj.length().get() for obj in self._partitions_cache.T[0]]
+                if len(self._partitions_cache.T) > 0
```

Comment:
Would it be more efficient to do `shape` instead of taking the length of the transpose?

Reply:
The transpose should just be metadata manipulation. I wrote it this way to make it easier to read/understand. There's no difference in the runtimes.